src/utils.ts

```diff
@@ -182,14 +186,28 @@ function getEntriesForEditor(editorPath: string) {
 }
 
 function generateEditorPath(appName: string) {
-  return join(
-    homedir(),
-    "Library",
-    "Application Support",
-    appName,
-    "User",
-    "History"
-  );
+  const platform = process.platform;
+
+  if (platform === "darwin") {
+    // macOS
+    return join(
+      homedir(),
+      "Library",
+      "Application Support",
+      appName,
+      "User",
+      "History"
+    );
+  } else if (platform === "linux") {
+    // Linux
+    return join(homedir(), ".config", appName, "User", "History");
+  } else if (platform === "win32") {
+    // Windows
+    return join(homedir(), "AppData", "Roaming", appName, "User", "History");
+  }
+
+  // Default fallback to Linux-style path
+  return join(homedir(), ".config", appName, "User", "History");
 }
```

Comment:
nit: this is a bit easier to follow

```suggestion
  if (platform === "darwin") {
    // macOS
    return join(
      homedir(),
      "Library",
      "Application Support",
      appName,
      "User",
      "History"
    );
  }
  
  if (platform === "win32") {
    // Windows
    return join(homedir(), "AppData", "Roaming", appName, "User", "History");
  }

  // Default fallback to Linux-style path
  return join(homedir(), ".config", appName, "User", "History");
}
```